students/Kristy_Martini/lesson03/mailroom.py

```diff
@@ -0,0 +1,111 @@
+import argparse
+import os
+
+class Report:
+    def __init__(self):
+        """ Report class holds the list of all donors and information associated with them."""
+        self.donors = list()
+        Kristy = Donor("Kristy Martini", 0, 3, 3000)
+        Mikey = Donor("Mike Martini", 0, 7, 3424834)
+        Cathy = Donor("Cathy Martini", 0, 7, 63833)
+        Bill = Donor("Bill Martini", 0, 10, 60000)
+        Nick = Donor("Nick Martini", 0, 50, 47484949)
+        
+        initial_donors = [Kristy, Mikey, Nick, Bill, Cathy]
+        for donor in initial_donors:
+            self.add_donor(donor)
+
+    def add_donor(self,Donor):
+        """ Add new donor to report"""
+        self.donors.append(Donor)
+
+    def create_report(self):
+        """ Display donor report to the user"""
+        print("Donor Name          | Total Given   | Num Gifts | Average Gift")
+        print("--------------------------------------------------------------")
+        for donor in self.donors:
+            name_sapce = 19 - len(donor.name)
+            total_space = 12 - len(str(donor.total_gift_value))
+            num_space = 10 - len(str(donor.num_gifts))
+            average_space = 10 - len(str(donor.average_gift))
+            print(donor.name, " "*name_sapce, "$", " "*total_space, donor.total_gift_value,
+                 " "*num_space, donor.num_gifts, "$", " "*average_space, donor.average_gift)
+        print("\n")
```

Comment:
Missing the sorting requirement - report should list donors by total given in descending order.